book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd

```diff
@@ -118,4 +118,56 @@ The `batch_run.R` script can take the following command-line arguments:
 
 [batch_run]: https://github.com/pecanproject/pecan/tree/develop/base/workflow/inst/batch_run.R
 [default_tests]: https://github.com/pecanproject/pecan/tree/develop/base/workflow/inst/default_tests.csv
-[xml_fun]: 
+[xml_fun]:
+
+### Continuous Integration
+
+Every time anyone commits a change to the PEcAn code, the act of pushing to GitHub triggers an automated build and test of the full PEcAn codebase using [Travis CI](https://travis-ci.org/pecanProject/pecan), and all pull requests must report a successful CI build before they will be merged. This will sometimes feel like a burden when the build breaks on an issue that looks trivial, but anything that breaks the build is important enough to fix. It's much better to find errors early and fix them before they get incorporated into the released PEcAn code.
+
+At this writing (September 2019), all our Travis builds run on the same version of Linux (currently Ubuntu 16.04) using three different versions of R in parallel: previous release, current release, and nightly builds of the R development branch. In most cases the build should pass on all three versions, but only the current release (`R:release`) is truly mandatory. We aim to fix errors found on R:release before merging, errors found on R:devel before the next R release, and errors found on R:oldrel as developer time and forward compatibility allow.
```

Comment:
I can never remember which Ubuntu version numbers correspond to which animal names, so might be helpful to add here (especially since it's in the Travis config).

```suggestion
At this writing (September 2019), all our Travis builds run on the same version of Linux (currently Ubuntu 16.04, `xenial`) using three different versions of R in parallel: previous release, current release, and nightly builds of the R development branch. In most cases the build should pass on all three versions, but only the current release (`R:release`) is truly mandatory. We aim to fix errors found on R:release before merging, errors found on R:devel before the next R release, and errors found on R:oldrel as developer time and forward compatibility allow.
```